lms/djangoapps/course_api/helpers.py

```diff
@@ -53,9 +53,14 @@ def get_marketing_data(course_key, language):
     marketing_root_format = marketing_link('COURSE_DETAILS_API_FORMAT')
     url = marketing_root_format.format(course_id=course_key)
 
-    response = requests.get(url=url, headers={
-        'Accept-Language': language,
-    })
+    try:
+        response = requests.get(
+            url=url,
+            headers={'Accept-Language': language},
+            timeout=3
+        )
+    except requests.exceptions.Timeout:
+        response = HttpResponseNotFound()
```

Comment:
Also add another check for all other `Exception` to make the API return more relevant information. I'd try it first myself to see the difference between the two behaviours.

```suggestion
        except Exception:
            log.exception('Something went wrong with the marketing courses API')
            raise APIException(
                {"status_code": status.HTTP_500_INTERNAL_SERVER_ERROR, "developerMessage": "Marketing courses didn't respond correctly."},
                code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            ) 

```

Reply:
Done